### PR TITLE
[2.x] Adds `toBeUrl()` Expectation

### PR DESCRIPTION
**Pull request by JonPurvis**
### What:

- [ ] Bug Fix
- [x] New Feature

### Description:

This PR adds a `toBeUrl()` expectation, to validate whether a value is a valid URL. This uses the same `isUrl()` method that Laravel has.

Example Usage:
```php
expect('https://pestphp.com')->toBeUrl();
expect('test')->not->toBeUrl();
```

### CODE REVIEW

File: src/Support/Str.php [hunk 1]
```diff
@@ -108,4 +129,22 @@ public static function describe(string $describeDescription, string $testDescrip
     {
         return sprintf('`%s` → %s', $describeDescription, $testDescription);
     }
+
+    /**
+     * Determine if a given value is a valid URL.
+     *
+     * @param  array<string>  $protocols
+     */
+    public static function isUrl(mixed $value, array $protocols = []): bool
```
Comment (nunomaduro) [inline src/Support/Str.php]: Can you remove `protocols` argument?

File: src/Support/Str.php [hunk 2]
```diff
@@ -108,4 +129,22 @@ public static function describe(string $describeDescription, string $testDescrip
...
+    {
+        if (! is_string($value)) {
```
Comment (nunomaduro) [inline src/Support/Str.php]: Please accept only a `string` on the parameter type.

File: src/Support/Str.php [hunk 3]
```diff
@@ -108,4 +129,22 @@ public static function describe(string $describeDescription, string $testDescrip
...
+            return false;
+        }
+
+        $protocolList = $protocols === []
```
Comment (nunomaduro) [inline src/Support/Str.php]: Can we don't have a fixed list of domains?

**Comment by JonPurvis** [comment]
Documentation PR: https://github.com/pestphp/docs/pull/237

**Review by owenvoke — APPROVED** [review]

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: src/Support/Str.php (hunk 1 above)
Comment (JonPurvis) [reply src/Support/Str.php]: Done!

File: src/Support/Str.php (hunk 2 above)
Comment (JonPurvis) [reply src/Support/Str.php]: Done!

File: src/Support/Str.php (hunk 3 above)
Comment (JonPurvis) [reply src/Support/Str.php]: I've made what I _think_ the change should be for this one

**Comment by JonPurvis** [comment]
Hey @nunomaduro 

Thanks for the feedback, I think I've actioned all of your points 😄

File: src/Support/Str.php [hunk 4]
```diff
@@ -20,6 +20,32 @@ final class Str
...
...ms-settings-displays-topology|ms-settings-emailandaccounts|ms-settings-language|ms-settings-location|ms-settings-lock|ms-settings-nfctransactions|ms-settings-notifications|ms-settings-power|ms-settings-privacy|ms-settings-proximity|ms-settings-screenrotation|ms-settings-wifi|ms-settings-workplace|ms-spd|ms-sttoverlay|ms-transit-to|ms-useractivityset|ms-virtualtouchpad|ms-visio|ms-walk-to|ms-whiteboard|ms-whiteboard-cmd|ms-word|msnim|msrp|msrps|mss|mtqp|mumble|mupdate|mvn|news|nfs|ni|nih|nntp|notes|ocf|oid|onenote|onenote-cmd|opaquelocktoken|openpgp4fpr|pack|palm|paparazzi|payto|pkcs11|platform|pop|pres|prospero|proxy|pwid|psyc|pttp|qb|query|redis|rediss|reload|res|resource|rmi|rsync|rtmfp|rtmp|rtsp|rtsps|rtspu|s3|secondlife|service|session|sftp|sgn|shttp|sieve|simpleledger|sip|sips|skype|smb|sms|smtp|snews|snmp|soap\.beep|soap\.beeps|soldat|spiffe|spotify|ssh|steam|stun|stuns|submit|svn|tag|teamspeak|tel|teliaeid|telnet|tftp|tg|things|thismessage|tip|tn3270|tool|ts3server|turn|turns|tv|udp|unreal|urn|ut2004|v-event|vemmi|ventrilo|videotex|vnc|view-source|wais|webcal|wpid|ws|wss|wtai|wyciwyg|xcon|xcon-userid|xfire|xmlrpc\.beep|xmlrpc\.beeps|xmpp|xri|ymsgr|z39\.50|z39\.50r|z39\.50s';
```
Comment (nunomaduro) [inline src/Support/Str.php]: Again, I don't want to maintain a list of all possible URLs in the world. Is not possible to use something like FILTER_VALIDATE_URL instead?

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: src/Support/Str.php (hunk 4 above)
Comment (JonPurvis) [reply src/Support/Str.php]: Ah yes, I'm with you now. Yeah, we can do that. I've just pushed an update that remove this and uses `FILTER_VALIDATE_URL`, same results, lets code to maintain 😄

**Comment by JonPurvis** [comment]
Thanks @nunomaduro 🥳

**Pull request merged**